crates/crane/src/typer.rs

```diff
@@ -51,20 +51,43 @@ struct ModuleItems {
 }
 
 pub struct Typer<'ctx> {
-    ctx: &'ctx TyContext,
+    ctx: &'ctx TyContext<'ctx>,
 
     modules: HashMap<TyPath, ModuleItems>,
     use_map: HashMap<TyPath, TyPath>,
     scopes: Vec<HashMap<TyPath, Arc<Type>>>,
+
+    // Types.
+    unit_ty: Arc<Type>,
```

Comment:
Added these back to the `Typer` for now until we're ready to use `Ty`s everywhere.